include/aie/Dialect/AIE/IR/AIEOps.td

```diff
@@ -1634,7 +1636,8 @@ def AIE_ObjectFifoCreateOp: AIE_Op<"objectfifo", [HasParent<"DeviceOp">, Symbol]
         TypeAttrOf<AIE_ObjectFifoType>:$elemType,
         BDDimLayoutArrayAttr:$dimensionsToStream,
         BDDimLayoutArrayArrayAttr:$dimensionsFromStreamPerConsumer,
-        DefaultValuedAttr<BoolAttr, "false">:$via_DMA
+        DefaultValuedAttr<BoolAttr, "false">:$via_DMA,
+        DefaultValuedAttr<BoolAttr, "false">:$plio
```

Comment:
This seems a bit odd to me..  Can't the objectFifo figure out that it's connected to PLIO based on the shimDMAAllocation op?  Could I have an objectFifo connecting 2 PLIOs?

Reply:
I am using this as a way for the user to specify whether the corresponding shim tile(s) of an object FIFO should be connected to GMIO or PLIO. Here is an example: https://github.com/Xilinx/mlir-aie/blob/plio-rebase/programming_examples/basic/passthrough_dmas_plio/aie2-input-plio.py#L49. 

We do also need to encode this in the shimDMAAllocationOP but that is later in the compilation, and I believe usually not exposed via the IRON python bindings. 

I believe an object FIFO could connect 2 PLIOs but have never tested it. It can connect two GMIOs, correct?

Reply:
The `shimDMAAllocationOp` is currently being generated by the AIEObjectFifoStatefulTransform and as the object FIFO also abstracts the `AIEFlowOps` between tiles I think it can be confusing to know where to add new attributes to guide the configuration. A cleaner solution might be to "allow" `shimDMAAllocationOps` to exist before the object FIFO lowering pass and have the lowering pass check for them and either: adopt whatever configuration they specify for the shim tile (including the channel index), or completing any missing information if it was not provided (following the same generation it currently uses). Another possibility would be to rely on the current `AIEFlowOp` in the same manner as above for a more general solution that would allow users to specify the ports they want for any tile not just the shim tile.

Reply:
Either works for me the only important thing to me is that this is user facing. That being said, that change in allowing `shimDMAAllocationOps` to be before the object FIFO lowering pass sounds like it could require some refactoring. If it is non-trivial, should that be a separate PR? If so, I can create an issue so we can track it.

Reply:
If we go this route, it will indeed require a bit of refactoring and I do think it will be better to have it in a separate PR that this PR would then build on. Let's discuss further and see if any of those solution would be preferred.

Reply:
> A cleaner solution might be to "allow" shimDMAAllocationOps to exist before the object FIFO lowering pass

+1 because it makes things more composable